packages/x-scheduler/src/material/timeline/content/view-header/MonthsHeader.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import { useStore } from '@base-ui-components/utils/store/useStore';
+import { getAdapter } from '../../../../primitives/utils/adapter/getAdapter';
+import { useMonthList } from '../../../../primitives/use-month-list';
+import { selectors } from '../../../../primitives/use-timeline';
+import { useTimelineStoreContext } from '../../../../primitives/utils/useTimelineStoreContext';
+
+import './Headers.css';
+
+const adapter = getAdapter();
+
+export function MonthsHeader() {
+  const getMonthList = useMonthList();
+  const store = useTimelineStoreContext();
+
+  const visibleDate = useStore(store, selectors.visibleDate);
+
+  const months = React.useMemo(
+    () =>
+      getMonthList({
+        date: visibleDate,
+        amount: 12,
+      }),
+    [getMonthList, visibleDate],
+  );
+
+  return (
+    <div className="MonthsHeader">
+      {months.map((month, index) => {
+        const monthNumber = adapter.getMonth(month);
+        return (
+          <React.Fragment key={`${adapter.format(month, 'monthShort')}-${adapter.getYear(month)}`}>
```

Comment:
In `useMonthList` you can use `processDate` like on `useDayList`
And then here you can use `month.key`